ppr-ui/src/components/collateral/generalCollateral/GenColAmend.vue

```diff
@@ -207,6 +207,9 @@ export default defineComponent({
         } else {
           newGeneralCollateral.push(amendedGC)
         }
+        if (localState.addDesc === '' && localState.delDesc === '') { // remove empty collateral altogether
+          newGeneralCollateral.pop()
```

Comment:
What are we removing here from `newGeneralCollateral`?
ie what is the last element in the array? How does this work?

Reply:
Yes, we are removing the last added Gen Col, if it has empty values.

In the logic above, we always push a new GC, even with empty values. And since there are multiple items in `generalCollateral` array, it allows to Undo (which we don't want for empty GC). So the logic here is to check if both Add and Del are empty, we simply remove the newly added GC.

<img width="461" alt="Screen Shot 2022-06-29 at 2 16 54 PM" src="https://user-images.githubusercontent.com/2333290/176547114-6df3f462-b9ab-48d5-8ce0-ab1ab2273fe6.png">

Reply:
Gotcha! Cool, thanks for the explanation sir 👍